include/sound/sof/header2.h

```diff
@@ -0,0 +1,410 @@
+#define SOF_IPC4_GLB_PIPE_MEM_SIZE(x)		((x) << SOF_IPC4_GLB_PIPE_MEM_SIZE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_EXT_LP_SHIFT		0
+#define SOF_IPC4_GL_PIPE_EXT_LP_MASK		BIT(0)
+#define SOF_IPC4_GL_PIPE_EXT_LP(x)		((x) << SOF_IPC4_GL_PIPE_EXT_LP_SHIFT)
+
+/* pipeline set state ipc msg */
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT	24
+#define SOF_IPC4_GL_PIPE_STATE_TYPE_MASK	GENMASK(28, 24)
+#define SOF_IPC4_GL_PIPE_STATE_TYPE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_TYPE_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_ID_SHIFT		16
+#define SOF_IPC4_GL_PIPE_STATE_ID_MASK		GENMASK(23, 16)
+#define SOF_IPC4_GL_PIPE_STATE_ID(x)		((x) << SOF_IPC4_GL_PIPE_STATE_ID_SHIFT)
+
+#define SOF_IPC4_GL_PIPE_STATE_SHIFT		0
+#define SOF_IPC4_GL_PIPE_STATE_MASK		GENMASK(15, 0)
+#define SOF_IPC4_GL_PIPE_STATE(x)		((x) << SOF_IPC4_GL_PIPE_STATE_SHIFT)
+
+enum module_type {
+	MOD_INIT_INSTANCE		= 0,
+	MOD_CONFIG_GET			= 1,
+	MOD_CONFIG_SET			= 2,
+	MOD_LARGE_CONFIG_GET		= 3,
+	MOD_LARGE_CONFIG_SET		= 4,
+	MOD_BIND			= 5,
+	MOD_UNBIND			= 6,
+	MOD_SET_DX			= 7,
+	MOD_SET_D0IX			= 8,
+	MOD_ENTER_MODULE_RESTORE	= 9,
+	MOD_EXIT_MODULE_RESTORE		= 10,
+	MOD_DELETE_INSTANCE		= 11,
+};
+
+enum sampling_frequency {
+	FS_8000HZ	= 8000,
+	FS_11025HZ	= 11025,
+	FS_12000HZ	= 12000, /* Mp3, AAC, SRC only. */
+	FS_16000HZ	= 16000,
+	FS_18900HZ	= 18900, /* SRC only for 44100 */
+	FS_22050HZ	= 22050,
+	FS_24000HZ	= 24000, /* Mp3, AAC, SRC only. */
+	FS_32000HZ	= 32000,
+	FS_37800HZ	= 37800, /* SRC only for 44100 */
+	FS_44100HZ	= 44100,
+	FS_48000HZ	= 48000, /* Default. */
+	FS_64000HZ	= 64000, /* AAC, SRC only. */
+	FS_88200HZ	= 88200, /* AAC, SRC only. */
+	FS_96000HZ	= 96000, /* AAC, SRC only. */
+	FS_176400HZ	= 176400, /* SRC only. */
+	FS_192000HZ	= 192000, /* SRC only. */
+	FS_INVALID
+};
+
+enum bit_depth {
+	DEPTH_8BIT	= 8, /* 8 bits depth */
+	DEPTH_16BIT	= 16, /* 16 bits depth */
+	DEPTH_24BIT	= 24, /* 24 bits depth - Default */
+	DEPTH_32BIT	= 32, /* 32 bits depth */
+	DEPTH_64BIT	= 64, /* 64 bits depth */
+	DEPTH_INVALID
+};
+
+enum channel_config {
+	/* one channel only. */
+	CHANNEL_CONFIG_MONO			= 0,
+	/* L & R. */
+	CHANNEL_CONFIG_STEREO			= 1,
+	/* L, R & LFE; PCM only. */
+	CHANNEL_CONFIG_2_POINT_1		= 2,
+	/* L, C & R; MP3 & AAC only. */
+	CHANNEL_CONFIG_3_POINT_0		= 3,
+	/* L, C, R & LFE; PCM only. */
+	CHANNEL_CONFIG_3_POINT_1		= 4,
+	/* L, R, Ls & Rs; PCM only. */
+	CHANNEL_CONFIG_QUATRO			= 5,
+	/* L, C, R & Cs; MP3 & AAC only. */
+	CHANNEL_CONFIG_4_POINT_0		= 6,
+	/* L, C, R, Ls & Rs. */
+	CHANNEL_CONFIG_5_POINT_0		= 7,
+	/* L, C, R, Ls, Rs & LFE. */
+	CHANNEL_CONFIG_5_POINT_1		= 8,
+	/* one channel replicated in two. */
+	CHANNEL_CONFIG_DUAL_MONO		= 9,
+	/* Stereo (L,R) in 4 slots, 1st stream: [ L, R, -, - ] */
+	CHANNEL_CONFIG_I2S_DUAL_STEREO_0	= 10,
+	/* Stereo (L,R) in 4 slots, 2nd stream: [ -, -, L, R ] */
+	CHANNEL_CONFIG_I2S_DUAL_STEREO_1	= 11,
+	/* L, C, R, Ls, Rs & LFE., LS, RS */
+	CHANNEL_CONFIG_7_POINT_1		= 12,
+	CHANNEL_CONFIG_INVALID
```

Comment:
what on earth is this about?

Again prefixes needed!

Reply:
ack